src/components/more-less-text/MoreLessText.tsx

```diff
@@ -0,0 +1,49 @@
+import Button from '@mui/material/Button';
+import Typography from '@mui/material/Typography';
+import { useState } from 'react';
+
+export const MoreLessText = ({ text, maxChars }: { text: string; maxChars: number }) => {
+  const [expanded, setExpanded] = useState(false);
+  const shouldTruncate = text.length > maxChars;
+  const reviewToDisplay = expanded ? text : truncateReview(text);
+
+  function truncateReview(text: string) {
```

Comment:
Thank you for noticing. You are right! truncateReview function shouldn't be inside of the component. I modified it a little and moved to the outer scope of the component but not to the utilts, since I think this function may not be really reusable for the other components.

Do you agree or it would be better to move it anyway?

Reply:
On the one hand, It is easy to make it generic enough to put it in utils

rename to `truncateText`, add `maxChars` param with default value

if we had unit tests, it would be the preferred way

on the other hand, I usually stick to the approach until I have at least 2 places with the same code, there is no need to do any abstraction

in this specific case, I think the first approach would be better